src/credentials/SsiCredentialIssuer.Reissuance.App/Handler/CredentialIssuerHandler.cs

```diff
@@ -0,0 +1,100 @@
+using Org.Eclipse.TractusX.SsiCredentialIssuer.DBAccess;
+using Org.Eclipse.TractusX.SsiCredentialIssuer.DBAccess.Repositories;
+using Org.Eclipse.TractusX.SsiCredentialIssuer.Entities.Enums;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+
+namespace Org.Eclipse.TractusX.SsiCredentialIssuer.Reissuance.App.Handlers;
+
+/// <inheritdoc />
+public class CredentialIssuerHandler : ICredentialIssuerHandler
+{
+    private readonly CredentialSettings _settings;
+    private readonly IServiceScopeFactory _serviceScopeFactory;
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    /// <param name="serviceScopeFactory"></param>
+    /// <param name="options"></param>
+    public CredentialIssuerHandler(IServiceScopeFactory serviceScopeFactory, IOptions<CredentialSettings> options)
+    {
+        _serviceScopeFactory = serviceScopeFactory;
+        _settings = options.Value;
+    }
+
+    /// <inheritdoc />
+    public async Task HandleCredentialProcessCreation(IssuerCredentialRequest credentialRequest)
+    {
+        var documentContent = Encoding.UTF8.GetBytes(credentialRequest.Schema);
+        var hash = SHA512.HashData(documentContent);
+        using var processServiceScope = _serviceScopeFactory.CreateScope();
+        var repositories = processServiceScope.ServiceProvider.GetRequiredService<IIssuerRepositories>();
+        var documentRepository = repositories.GetInstance<IDocumentRepository>();
+        var reissuanceRepository = repositories.GetInstance<IReissuanceRepository>();
+        var companyCredentialDetailsRepository = repositories.GetInstance<ICompanySsiDetailsRepository>();
+        var docId = documentRepository.CreateDocument($"{credentialRequest.TypeId}.json", documentContent,
+            hash, MediaTypeId.JSON, DocumentTypeId.PRESENTATION, x =>
+            {
+                x.IdentityId = credentialRequest.IdentiyId;
+                x.DocumentStatusId = DocumentStatusId.ACTIVE;
+            }).Id;
+
+        Guid? processId = CreateProcess(repositories);
+
+        var ssiDetailId = companyCredentialDetailsRepository.CreateSsiDetails(
+            credentialRequest.Bpnl,
+            credentialRequest.TypeId,
+            CompanySsiDetailStatusId.ACTIVE,
+            _settings.IssuerBpn,
+            credentialRequest.IdentiyId,
+            c =>
+            {
+                c.VerifiedCredentialExternalTypeDetailVersionId = credentialRequest.DetailVersionId;
+                c.ProcessId = processId;
+                c.ExpiryDate = credentialRequest.ExpiryDate;
+            }).Id;
+
+        documentRepository.AssignDocumentToCompanySsiDetails(docId, ssiDetailId);
+
+        reissuanceRepository.CreateReissuanceProcess(credentialRequest.Id, ssiDetailId);
+
+        companyCredentialDetailsRepository.CreateProcessData(ssiDetailId, JsonDocument.Parse(credentialRequest.Schema), credentialRequest.KindId,
```

Comment:
this will not work with the current process. It will fail when trying to import the created credential to the holder wallet, because we need the auth data for that. In the current api implementation when creating the credential the auth data are send within the request of the creation.